### PR TITLE
Interproscan assembly pipeline

### PR DESCRIPTION
**Pull request by chrisAta**
Updated existing interproscan module so that you can mount a interproscan_db directory directly. uses the same container as the Mettannotator interproscan module to achieve this. Also updated the unit tests to nf-test from pytest

### CODE REVIEW

File: modules/ebi-metagenomics/interproscan/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// This krona/ktimporttext unit test file is converted to nf-test from an existing pytest test at this commit: https://github.com/nf-core/modules/commit/5e34754d42cd2d5d248ca8673c0a53cdf5624905
+// I have converted it because the standard way of writing nf-core unit tests now is to use nf-test
+// It is otherwise exactly the same unit test
```
Comment (vagkaratzas) [inline modules/ebi-metagenomics/interproscan/tests/main.nf.test:3]: Remove leftover comments up here

File: modules/ebi-metagenomics/interproscan/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    tag "interproscan"
+
+    test("test_interproscan") {
+
+        when {
+            process {
+                """                
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                    file("${moduleDir}/tests/test.fasta", checkIfExists: true)
```
Comment (vagkaratzas) [inline modules/ebi-metagenomics/interproscan/tests/main.nf.test:23]: let's move the test file (test.fasta) to a folder named 'fixtures'? Probably the interproscan_db as well? Keep the folder architecture more or less the same across modules.

File: modules/ebi-metagenomics/interproscan/main.nf [hunk 3]
```diff
@@ -25,7 +30,7 @@ process INTERPROSCAN {
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
 
-    def appl = "-appl TIGRFAM,FunFam,SFLD,PANTHER,Gene3D,Hamap,ProSiteProfiles,Coils,SMART,CDD,PRINTS,PIRSR,ProSitePatterns,AntiFam,Pfam,MobiDBLite"
+    def appl = "-appl TIGRFAM,SFLD,SUPERFAMILY,Gene3D,Hamap,Coils,ProSiteProfiles,SMART,CDD,PRINTS,PIRSF,ProSitePatterns,PfamA,ProDom,MobiDBLite,SignalP_GRAM_POSITIVE,SignalP_GRAM_NEGATIVE,SignalP_EUK,Phobius,TMHMM"
     if ( args.contains("-appl") ) {
```
Comment (vagkaratzas) [inline modules/ebi-metagenomics/interproscan/main.nf:34]: Some of the -appl arguments were creating errors for me on Codon. If I remember correctly Java erros. Can you run a fast one on Codon with all enabled to make sure this is not the case any more?

**Review by vagkaratzas — CHANGES_REQUESTED** [review]
Excellent! Some minor things to check and discuss ;)

File: modules/ebi-metagenomics/interproscan/main.nf (hunk 3 above)
Comment (chrisAta) [reply modules/ebi-metagenomics/interproscan/main.nf:34]: Tested this with this version of the database: https://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/5.62-94.0/ 

no errors anymore

**Review by vagkaratzas — APPROVED** [review]
Nice. Feel free to merge when ready ;)

**Pull request merged**